odk/odk.py

```diff
@@ -980,6 +1073,63 @@ def dump_schema(class_name):
         clazz = globals()[class_name]  # Get the class object from the globals dictionary
         print(json.dumps(clazz.json_schema(), sort_keys=True, indent=4))
 
+@cli.command()
+@click.option('-T', '--templatedir', default='/tools/templates/')
+def update(templatedir):
+    """
+    Updates a pre-existing repository. This command is expected to be
+    run from within the src/ontology directory (the directory
+    containing the configuration file).
+    """
+    config_matches = list(glob.glob('*-odk.yaml'))
+    if len(config_matches) == 0:
+        raise click.ClickException("No ODK configuration file found")
+    elif len(config_matches) > 1:
+        raise click.ClickException("More than ODK configuration file found")
+    config = config_matches[0]
+    mg = Generator()
+    mg.load_config(config)
+    project = mg.context.project
+
+    # When updating, for most files, we only install them if
+    # they do not already exist in the repository (typically
+    # because they are new files that didn't exist in the
+    # templates of the previous version of the ODK). But a
+    # handful of files are not reinstalled even if they are
+    # missing (e.g. DOSDP example files) or on the contrary
+    # always reinstalled to overwrite any local changes (e.g.
+    # the main Makefile). We declare the corresponding policies.
+    policies = [
+            ('CODE_OF_CONDUCT.md', NEVER),
+            ('CONTRIBUTING.md', NEVER),
+            ('issue_template.md', NEVER),
```

Comment:
Should really be a structured template in the `.github` folder. I am digressing.

Reply:
Yes, with that PR we have the infrastructure that could allow us to give users a way to precisely control what they want to overwrite or not during an update.

But I think this would warrant more discussion and can be postponed to a future feature.